src/main/java/com/example/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,72 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.service.BookService;
+import com.example.onlinebookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create new category",
+            description = "Create new book category and save it to db")
+    public CategoryDto createCategory(@RequestBody CategoryDto categoryDto) {
+        return categoryService.save(categoryDto);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/{id}")
+    @Operation(summary = "Update category by id", description = "Update book category by id")
+    public CategoryDto updateCategory(@PathVariable Long id, @RequestBody CategoryDto categoryDto) {
```

Comment:
Validations won't be performed without relevant annotation near controller argument
Please recheck through project
```suggestion
    public CategoryDto updateCategory(@PathVariable Long id, @Valid @RequestBody CategoryDto categoryDto) {
```

Reply:
fixed